com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs

```diff
@@ -454,7 +457,29 @@ public override void Setup(ScriptableRenderContext context, ref RenderingData re
 
             // The copying of depth should normally happen after rendering opaques.
             // But if we only require it for post processing or the scene camera then we do it after rendering transparent objects
-            m_CopyDepthPass.renderPassEvent = (!requiresDepthTexture && (applyPostProcessing || isSceneViewCamera || isGizmosEnabled)) ? RenderPassEvent.AfterRenderingTransparents : RenderPassEvent.AfterRenderingOpaques;
+            // Aim to have the most optimized render pass event for Depth Copy (The aim is to minimize the number of render passes)
+            if (requiresDepthTexture)
+            {
+                RenderPassEvent copyDepthPassEvent = RenderPassEvent.AfterRenderingTransparents;
+                // CameraData's requiresDepthTexture is based on the URP Pipeline Asset UI checkbox value.
+                if (cameraData.requiresDepthTexture)
+                {
+                    // If user explicitly enable the setting, we force the depth copy to be AfterRenderingSkybox
+                    copyDepthPassEvent = (RenderPassEvent)Mathf.Min((int)copyDepthPassEvent, (int)RenderPassEvent.AfterRenderingSkybox);
```

Comment:
Isn't this the same as always setting copyDepthPassEvent to `RenderPassEvent.AfterRenderingSkybox`

Reply:
This might introduce the regression we talked before that moves the copy depth pass from afterRenderingOpaques -> afterRenderingSkybox.

I discussed with Andre a possible solution for this that would not only fix the copy  depth in between opaque and sky but also other causes of breaking render pass. The problem is the that default event is `AfterRenderingOpaques`, so users can inject other types of passes and break opaque/sky by mistake. I tagged you into a proposal fix.

Reply:
this can cause users code regression, will remove the first branch

Reply:
fixed in CS 79b78f0 as discussed